vividus-extension-selenium/src/main/java/org/vividus/steps/ui/GenericNestedSteps.java

```diff
@@ -114,12 +114,18 @@ public void performAllStepsWhileElementsExist(ComparisonRule comparisonRule, int
     private boolean isExpectedElementsQuantity(Locator locator, Matcher<Integer> elementsMatcher,
             MutableBoolean firstIteration)
     {
+        int foundElements = getElementsNumber(locator);
+        boolean elementsMatch;
         if (firstIteration.isTrue())
         {
             firstIteration.setValue(false);
-            return softAssert.assertThat("Elements number", getElementsNumber(locator), elementsMatcher);
+            elementsMatch = softAssert.assertThat("Elements number", foundElements, elementsMatcher);
         }
-        return elementsMatcher.matches(getElementsNumber(locator));
+        else
+        {
+            elementsMatch = elementsMatcher.matches(foundElements);
+        }
+        return elementsMatch && foundElements != 0;
```

Comment:
what if step conditions is `= 0`? I guess we should either add the validation of the condition or change the step syntax